src/p4est_io.h

```diff
@@ -58,19 +58,9 @@ SC_EXTERN_C_BEGIN;
 #define P4EST_NUM_USER_STRING_BYTES 48 /**< number of user string bytes */
 #define P4EST_NUM_FIELD_HEADER_BYTES (2 + P4EST_NUM_ARRAY_METADATA_BYTES + P4EST_NUM_USER_STRING_BYTES)
                                      /**< number of bytes of one field header */
-#define P4EST_FILE_COUNT_ERROR -1 /**< All other error codes are defined by MPI or are
-                                     error codes defined by libsc.
-                                     This error code is used to indicate a read
-                                     or write count error that may be occurred during a
-                                     MPI IO operation or a IO operation called by C
-                                     standard functions. */
-
-/** This macro is used for file format errors. */
-#ifndef P4EST_ENABLE_MPIIO
-#define P4EST_ERR_IO EIO /**< File format error code without MPI IO */
-#else
-#define P4EST_ERR_IO sc_MPI_ERR_IO /**< File format error with MPI IO */
-#endif
+#define P4EST_FILE_MAX_GLOBAL_QUAD 9999999999999999 /**< maximal number of global quadrants */
+#define P4EST_FILE_MAX_BLOCK_SIZE 9999999999999 /**< maximal number of block bytes */
+#define P4EST_FILE_MAX_FIELD_ENTRY_SIZE 9999999999999 /**< maximal numeber of bytes per field entry */
```

Comment:
Not to be too perfectionist, but would it be easier/safer to use something like (?):
`1000UL * 1000UL * 1000UL - 1UL`

Reply:
Yes, I think it would be easier to read and safer if we use `((10000UL * 10000UL * 10000UL * 10000UL) - 1UL)`

Reply:
I changed the macros to `((10000UL * 10000UL * 10000UL * 10000UL) - 1UL)` and `((10000L * 10000L * 10000L * 10000L) - 1L)`, respectively. Since the number of global quadrants in p4est is a signed int, I took this into account for the corresponding macro.